nomos-services/mempool/src/tx/service.rs

```diff
@@ -6,28 +6,71 @@ pub mod openapi {
 
 // std
 use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::path::PathBuf;
 
 // crates
 use futures::StreamExt;
+use overwatch_rs::services::state::ServiceState;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tracing::error;
+use utils::overwatch::recovery::backends::FileBackendSettings;
+use utils::overwatch::{JsonFileBackend, RecoveryOperator};
 // internal
-use crate::backend::MemPool;
+use crate::backend::{MemPool, RecoverableMempool};
 use crate::network::NetworkAdapter;
 use crate::{MempoolMetrics, MempoolMsg};
 use nomos_network::{NetworkMsg, NetworkService};
 use overwatch_rs::services::life_cycle::LifecycleMessage;
 use overwatch_rs::services::{
     handle::ServiceStateHandle,
     relay::{OutboundRelay, Relay},
-    state::{NoOperator, NoState},
     ServiceCore, ServiceData, ServiceId,
 };
-use tracing::error;
+
+#[derive(Error, Debug)]
+pub enum StateError {}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MempoolState<NS, PS, P> {
+    // `pool` is an option so that we don't need it to implement `Mempool`
+    // when implementing `ServiceState` for this type, as otherwise we would need to call `P::new(settings)`.
+    pool: Option<P>,
+    _phantom: PhantomData<(NS, PS)>,
+}
+
+// We need to have three generics instead of two as otherwise we would need `P` to implement `Mempool`
+// to access its `Settings` associated type, which would defeat declaring a new type for the recovery state.
+impl<NS, PS, P> From<P> for MempoolState<NS, PS, P> {
+    fn from(value: P) -> Self {
+        Self {
+            pool: Some(value),
+            _phantom: PhantomData,
+        }
+    }
+}
+
+impl<NS, PS, P> ServiceState for MempoolState<NS, PS, P> {
+    type Error = StateError;
+    type Settings = TxMempoolSettings<PS, NS>;
+
+    fn from_settings(_settings: &Self::Settings) -> Result<Self, Self::Error> {
+        Ok(Self {
+            pool: None,
+            _phantom: PhantomData,
+        })
+    }
+}
 
 pub struct TxMempoolService<N, P>
 where
     N: NetworkAdapter<Payload = P::Item, Key = P::Key>,
-    P: MemPool,
-    P::Settings: Clone,
+    N::Settings: Send,
+    P: RecoverableMempool,
+    P::RecoveryState: Clone + Send + Sync + Serialize + DeserializeOwned,
```

Comment:
Do we really need the clone here? I think it should be safe to remove it and just make the call in the RecoverableMempool implementation where is used.

Reply:
Ah yeah, its true that the state is `Clone`.  Maybe that is not really necessary :thinking: 
@AlejandroCabeza did we introduce that during the latest changes or was it like that originally? I think that bound should not really be necessary.

Reply:
I can check whether it's required or not. In general, I am personally not a big fan of having bounds on trait associated types, unless they are required in one of the methods exposed by the trait. Otherwise, those bounds can easily be expressed in the place where the trait is used.

Reply:
Yeah I'll give it a crack.

Reply:
@danielSanchezQ @AlejandroCabeza I got rid of most of the aforementioned bounds in this PR: https://github.com/logos-co/Overwatch/pull/46. The final result that I can see from the examples and tests in the overwatch repo does not look any worse. The only thing I cannot fully picture yet, is whether removing the inter-trait bounds would make services like cryptarchia more complicated, but I don't think that will be the case. I think the new structure would make simpler services easier to define. Let me know what you guys think!

Reply:
@danielSanchezQ I feel like we didn't introduce them recently but I might be wrong. IIRC the Clone behaviour was there to be able to share States among services.